### PR TITLE
Update examples for latest `crux_http`

### PR DESCRIPTION
**Pull request by StuartHarris**
- [x] update crate deps
- [x] expose `HttpErrorHttp` from `crux_http`
- [x] add `impl From<Result<HttpResponse, HttpError>> for HttpResult` to `crux_http`
- [x] update all variants of `counter` and `cat_facts` examples to use `HttpResult`
- [x] move `cli` and `tauri` examples from `surf` to `reqwest` (`surf` now requires silly conversions since we started using a fork of `http-types` in `crux_http`)
- [x] move from `async-std` to `tokio`

### CODE REVIEW

**Comment by StuartHarris** [comment]
Apologies for the large PR — mostly deps and example updates. The only thing that really needs reviewing are the changes to `crux_http` before we publish a new version with some small changes to help with errors

File: crux_http/src/lib.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ pub use http_types::{self as http};
 
 pub use self::{
     config::Config,
-    error::HttpError,
+    error::{HttpError, HttpErrorHttp},
```
Comment (obmarg) [inline crux_http/src/lib.rs:29]: Doesn't look like it was introduced by this PR but `HttpErrorHttp` seems like a bad name for a type....

**Review by obmarg — APPROVED** [review]

File: crux_http/src/lib.rs (hunk 1 above)
Comment (charypar) [reply crux_http/src/lib.rs:29]: We could make that into a struct variant on the `HttpError` and move the impl into `HttpError` I suppose 🤔 That's why it's called it - it's a struct for the `Http` variant of the `HttpError`

Comment (obmarg) [reply crux_http/src/lib.rs:29]: Hmm, yes, I see.  Well 🤷 I have no real preference, just saw the name and thought "what is going on". 

Will leave yous to decide whether or not it's fine.

Comment (StuartHarris) [reply crux_http/src/lib.rs]: I made it a struct variant. Seems much cleaner

**Pull request merged**